### ISSUE TITLE
bootctl pager corrupt "source:" output under "Default Boot Loader Entry:"

### ISSUE THREAD

**Issue by jameshilliard** [systemd/systemd#20195]
**systemd version the issue has been seen with**
 > systemd 249

**Used distribution**
 > buildroot

**Linux kernel version used** (`uname -a`)

 > `Linux buildroot 5.13.0 #1 SMP Sat Jul 10 09:51:13 UTC 2021 x86_64 GNU/Linux`

**CPU architecture issue was seen on**
 > x86_64

**Expected behaviour you didn't see**
 > normal output in `bootctl status`: `source: /boot/loader/entries/alt.conf`

**Unexpected behaviour you saw**
 > corrupt output in `bootctl status`: `source: ]8;;file://buildroot/boot/loader/entries/alt.confG/boot/loader/entries/alt.conf]8;;G`

**Steps to reproduce the problem**
 > run `bootctl status`

**Additional program output to the terminal or log subsystem illustrating the issue**

corrupt output appears when using `bootctl status` but not `bootctl status --no-pager`, so bug appears to be related to the pager

**Comment by jameshilliard** [comment]
Ok, so this doesn't appear to be limited to  bootctl
```text
# systemctl status network.service
● network.service - Network Connectivity
     Loaded: loaded (]8;;file://buildroot/etc/systemd/system/network.serviceG/etc/systemd/system/network.service]8;;G; enabled; vendor preset: enabled)
     Active: active (exited) since Sat 2021-07-10 11:16:31 UTC; 19h ago
    Process: 209 ExecStart=/sbin/ip addr flush dev lo (code=exited, status=0/SUCCESS)
    Process: 226 ExecStart=/sbin/ifup -a (code=exited, status=0/SUCCESS)
   Main PID: 226 (code=exited, status=0/SUCCESS)
     CGroup: /system.slice/network.service
             └─346 udhcpc -R -n -O search -p /var/run/udhcpc.enp2s0.pid -i enp2s0 -x hostname:buildroot

Notice: journal has been rotated since unit was started, output may be incomplete.
```
```text
# systemctl status network.service --no-pager
● network.service - Network Connectivity
     Loaded: loaded (/etc/systemd/system/network.service; enabled; vendor preset: enabled)
     Active: active (exited) since Sat 2021-07-10 11:16:31 UTC; 19h ago
    Process: 209 ExecStart=/sbin/ip addr flush dev lo (code=exited, status=0/SUCCESS)
    Process: 226 ExecStart=/sbin/ifup -a (code=exited, status=0/SUCCESS)
   Main PID: 226 (code=exited, status=0/SUCCESS)
     CGroup: /system.slice/network.service
             └─346 udhcpc -R -n -O search -p /var/run/udhcpc.enp2s0.pid -i enp2s0 -x hostname:buildroot

Notice: journal has been rotated since unit was started, output may be incomplete.
```

**Comment by mrc0mmand** [comment]
I suspect this is the same issue as https://github.com/systemd/systemd/issues/18814 - i.e. you have less < 563 which doesn't support links.

**Comment by jameshilliard** [comment]
> you have less < 563

Nope, it's busybox less:
```text
# less
BusyBox v1.33.1 (2021-07-08 10:42:54 UTC) multi-call binary.

Usage: less [-EFIMmNSRh~] [FILE]...

View FILE (or stdin) one screenful at a time

	-E	Quit once the end of a file is reached
	-F	Quit if entire file fits on first screen
	-I	Ignore case in all searches
	-M,-m	Display status line with line numbers
		and percentage through the file
	-N	Prefix line number to each line
	-S	Truncate long lines
	-R	Remove color escape codes in input
	-~	Suppress ~s displayed past EOF
```

**Comment by jameshilliard** [comment]
Looks like we probably need a meson flag to disable this incompatible behavior?

**Comment by mrc0mmand** [comment]
> > you have less < 563
> 
> Nope, it's busybox less:
> 
> ```
> # less
> BusyBox v1.33.1 (2021-07-08 10:42:54 UTC) multi-call binary.
> 
> Usage: less [-EFIMmNSRh~] [FILE]...
> 
> View FILE (or stdin) one screenful at a time
> 
> 	-E	Quit once the end of a file is reached
> 	-F	Quit if entire file fits on first screen
> 	-I	Ignore case in all searches
> 	-M,-m	Display status line with line numbers
> 		and percentage through the file
> 	-N	Prefix line number to each line
> 	-S	Truncate long lines
> 	-R	Remove color escape codes in input
> 	-~	Suppress ~s displayed past EOF
> ```

Which doesn't seem to support the link feature (I'm not completely sure which version of less it is based on):
```
$ less --version
less 581.2 (POSIX regular expressions)
...
$ PAGER="less" systemctl status systemd-journald
● systemd-journald.service - Journal Service
     Loaded: loaded (/usr/lib/systemd/system/systemd-journald.service; static)
     Active: active (running) since Sat 2021-07-10 09:48:55 CEST; 1 day 2h ago
TriggeredBy: ● systemd-journald-dev-log.socket
             ● systemd-journald.socket
             ● systemd-journald-audit.socket
       Docs: man:systemd-journald.service(8)
             man:journald.conf(5)
...
```
```
$ busybox less
BusyBox v1.33.1 (2021-05-06 18:08:02 UTC) multi-call binary.
...
$ ln -s `which busybox` ~/bin/less
$ PATH="$HOME/bin:$PATH" PAGER="less" systemctl status systemd-journal
● systemd-journald.service - Journal Service
     Loaded: loaded (]8;;file://pyrelight/usr/lib/systemd/system/systemd-journald.serviceG/usr/lib/systemd/system/systemd-journald.service]8;;G; static)
     Active: active (running) since Sat 2021-07-10 09:48:55 CEST; 1 day 2h ago
TriggeredBy: ● systemd-journald-dev-log.socket
             ● systemd-journald.socket
             ● systemd-journald-audit.socket
       Docs: ]8;;man:systemd-journald.service(8)Gman:systemd-journald.service(8)]8;;G
             ]8;;man:journald.conf(5)Gman:journald.conf(5)]8;;G
...
```



> Looks like we probably need a meson flag to disable this incompatible behavior?

/cc @poettering

**Comment by jameshilliard** [comment]
> I'm not completely sure which version of less it is based on

I think busybox versions of utilities like less are usually fully independent implementations.

Added a flag to disable urlify in #20196.

**Comment by jameshilliard** [comment]
Fixed by #20196.
